libvast/src/meta_index.cpp

```diff
@@ -27,41 +27,31 @@
 namespace vast {
 
 void meta_index::add(const uuid& partition, const table_slice& slice) {
-  auto& part_synopsis = partition_synopses_[partition];
-  auto& layout = slice.layout();
-  if (blacklisted_layouts_.count(layout) == 1)
-    return;
-  auto i = part_synopsis.find(layout);
-  table_synopsis* table_syn;
-  if (i != part_synopsis.end()) {
-    table_syn = &i->second;
-  } else {
-    // Create new synopses for a layout we haven't seen before.
-    i = part_synopsis.emplace(layout, table_synopsis{}).first;
-    table_syn = &i->second;
-    for (auto& field : layout.fields) {
-      auto syn = has_skip_attribute(field.type)
-                   ? nullptr
-                   : factory<synopsis>::make(field.type, synopsis_options_);
-      if (table_syn->emplace_back(std::move(syn)) != nullptr)
-        VAST_DEBUG(this, "created new synopsis structure for type", field.type);
-    }
-    // If we couldn't create a single synopsis for the layout, we will no
-    // longer attempt to create synopses in the future.
-    auto is_nullptr = [](auto& x) { return x == nullptr; };
-    if (std::all_of(table_syn->begin(), table_syn->end(), is_nullptr)) {
-      VAST_DEBUG(this, "could not create a synopsis for layout:", layout);
-      blacklisted_layouts_.insert(layout);
-    }
-  }
-  VAST_ASSERT(table_syn->size() == slice.columns());
-  for (size_t col = 0; col < slice.columns(); ++col)
-    if (auto& syn = (*table_syn)[col])
+  auto make_synopsis = [&](const record_field& field) -> synopsis_ptr {
+    return has_skip_attribute(field.type)
+             ? nullptr
+             : factory<synopsis>::make(field.type, synopsis_options_);
+  };
+  auto& part_syn = synopses_[partition];
+  auto add_to_blacklist = true;
```

Comment:
`add_to_blacklist` is only assigned to, but never used. The blacklist itself has been removed from the type, so this variable is a leftover?

Reply:
A leftover, indeed.